route.rb

```diff
@@ -34,4 +37,11 @@ def start_station
   def end_station
     @stations[-1]
   end
+
+  protected
+
+  def validate!
+    raise 'Неверный класс входных данных' if !@stations.all?{|station| station.class == Station}
```

Comment:
и в моем комментарии и в документации есть вот такие лаконичные варианты:
```suggestion
    raise 'Неверный класс входных данных' if !@stations.all?(Station)
```

еще лучше через `unless`:
```ruby
    raise 'Неверный класс входных данных' unless @stations.all?(Station)
```